lib/rules/no-return-assign.js

```diff
@@ -36,11 +36,30 @@ function isEnclosedInParens(node, context) {
 
 module.exports = function(context) {
     var always = (context.options[0] || "except-parens") !== "except-parens";
+    var message;
+
+    /**
+     * Check whether return statement contains assignment
+     * @param {ASTNode} nodeToCheck node to check
+     * @param {ASTNode} nodeToReport node to report
+     * @returns {void}
+     * @private
+     */
+    function checkForAssignInReturn(nodeToCheck, nodeToReport) {
+        if (isAssignment(nodeToCheck) && (always || !isEnclosedInParens(nodeToCheck, context))) {
+            context.report(nodeToReport, message);
+        }
+    }
 
     return {
         "ReturnStatement": function(node) {
-            if (isAssignment(node.argument) && (always || !isEnclosedInParens(node.argument, context))) {
-                context.report(node, "Return statement should not contain assignment.");
+            message = "Return statement should not contain assignment.";
+            checkForAssignInReturn(node.argument, node, message);
```

Comment:
This passes in `message` as the third argument to `checkForAssignInReturn`, which only takes two arguments and pulls `message` from the closure. Perhaps pass the argument instead of using the closure variable?